### PR TITLE
Symlinks - create symbolic links to target directory

### PR DESCRIPTION
**Pull request by wanyufe**
*Issue #, if available:*
Root disk size is limited. Symlinks can create symbolic links in root folder and target against directory in data disk.
*Description of changes:*
A new map[string]string is added in machineconfig, so that symbolic link location (key) and targe location (value) can be stored. 

The actual `ln -s`  commands is in eksa prekubeadmCommands which will be mapped to cloud-init runcmd.
*Testing performed:*
Unit testing and manual testing.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: api/v1beta1/cloudstackmachine_webhook.go [hunk 1]
```diff
@@ -68,6 +69,14 @@ func (r *CloudStackMachine) ValidateCreate() error {
 	if len(r.Spec.DiskOffering.ID) > 0 || len(r.Spec.DiskOffering.Name) > 0 {
 		errorList = webhookutil.EnsureIntFieldsAreNotNegative(r.Spec.DiskOffering.CustomSize, "customSizeInGB", errorList)
 	}
+	for symlink, target := range r.Spec.Symlinks {
+		if !strings.HasPrefix(symlink, "/") || strings.HasSuffix(symlink, "/") {
+			errorList = append(errorList, field.Invalid(field.NewPath("spec", "Symlinks"), symlink, "symlink (key) must start with / and NOT ends with /"))
```
[the remark this answers is not in the document]
Comment (wanyufe) [reply api/v1beta1/cloudstackmachine_webhook.go:74]: thanks for catching the typo. 

the actual error message looks like: 
```
Message: "Invalid value: \"/var/log/pods/\": symlink (key) must start with / and NOT end with /",
```

**Comment by rejoshed** [comment]
@wanyufe 

> A new map[string]string is added in machineconfig, so that symbolic link location (key) and targe location (value) can be stored.

I'm not sure I see the need for this field.

From my read, it's not used in any controller, so it's just an additional optional field for EKS-A's consumption?

If my read is correct, then  to me, this should be handled on the EKS-A side entirely via a k8s [Annotation](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) on the machine templates.

Please let me know your thoughts on using annotations instead and the motivations for a new field in CAPC?

Otherwise, the code overall is fine.

Thanks!

**Comment by wanyufe** [comment]
@rejoshed 
I like the idea of using Annotation instead of changing/adding the CAPC interface. 
I would like to talk with eksa team about this idea and it definitely needs some changes in eksa code to retrieve those annotation and construct eksa struct -- I think this is a good move for eks-a in a long run.

The changes should also include other similar fields we added recently, like mount/filesystem/device/label in disk offering PR (already merged) and future PR( iso attachment).

So I suggest make these changes into a separate PR, what do you think?

**Comment by wanyufe** [comment]
closing PR:

symlinks will be purely implemented in eksa, and cloudstackmachine template annotation will be used to store the related info.

So no changes to capc project.

**Pull request closed without merging**